src/lib/pipeline.js

```diff
@@ -334,6 +334,12 @@ class Pipeline extends Builder {
     });
   }
 
+  stop() {
+    return Promise.all(
+      this.inputs.filter(input => input.stop).map(input => input.stop())
```

Comment:
Logically `forEach` would make more sense than `map` here (and I believe it performs better)